R/score_transition_risk_and_polish.R

```diff
@@ -0,0 +1,208 @@
+    unnest_product(emissions_profile),
+    unnest_product(sector_profile)
+  )
+
+  if (!pivot_wider) {
+    hint <- "Do you need `options(tiltIndicatorAfter.output_co2_footprint = TRUE)`?"
+    unnest_product(emissions_profile) |> check_col("co2_footprint", hint)
+  }
+
+  select_emissions_profile_product <- unnest_product(emissions_profile) |>
+    select(
+      c(
+        "companies_id",
+        "country",
+        "main_activity",
+        "ep_product",
+        "activity_uuid_product_uuid",
+        "matched_activity_name",
+        "matched_reference_product",
+        "unit",
+        "co2e_lower",
+        "co2e_upper",
+        "emission_profile",
+        "benchmark",
+        "profile_ranking",
+        "tilt_sector",
+        "tilt_subsector",
+        if (!pivot_wider) "co2_footprint"
+      )
+    )
+  select_sector_profile_product <- unnest_product(sector_profile) |>
+    select(
+      c(
+        "companies_id",
+        "ep_product",
+        "sector_profile",
+        "scenario",
+        "year",
+        "reduction_targets"
+      )
+    )
+
+  select_transition_risk_score_product <- unnest_product(transition_risk_score) |>
+    select(c(
+      "companies_id",
+      "ep_product",
+      "benchmark_tr_score",
+      "transition_risk_score"
+    ))
+
+  out_product <- select_emissions_profile_product |>
+    left_join(
+      select_sector_profile_product,
+      relationship = "many-to-many",
+      by = c("companies_id", "ep_product")
+    ) |>
+    unite("benchmark_tr_score", all_of(benchmark_cols()), remove = FALSE) |>
+    left_join(
+      select_transition_risk_score_product,
+      by = c("companies_id", "ep_product", "benchmark_tr_score")
+    ) |>
+    distinct()
+
+  select_emissions_profile_company <- unnest_company(emissions_profile) |>
+    select(
+      c(
+        "companies_id",
+        "country",
+        "main_activity",
+        "co2e_lower",
+        "co2e_upper",
+        "benchmark",
+        "emission_profile",
+        "emission_profile_share",
+        "profile_ranking_avg",
+        if (!pivot_wider) "co2_avg"
+      )
+    )
+
+  select_sector_profile_company <- unnest_company(sector_profile) |>
+    select(c("companies_id", "scenario", "year", "reduction_targets_avg"))
+
+  select_transition_risk_score_company <- unnest_company(transition_risk_score) |>
+    select(
+      c(
+        "companies_id",
+        "benchmark_tr_score_avg",
+        "transition_risk_score_avg"
+      )
+    )
+
+  tmp <- select_emissions_profile_company
+  if (pivot_wider) {
+    tmp <- tmp |>
+      exclude_cols_then_pivot_wider(
+        exclude_cols = "co2e",
+        id_cols = c(
+          "companies_id",
+          "country",
+          "main_activity",
+          "benchmark",
+          "profile_ranking_avg",
+          if (!pivot_wider) "co2_avg"
+        ),
+        names_from = "emission_profile",
+        names_prefix = "emission_category_",
+        values_from = "emission_profile_share"
+      )
+  }
```

Comment:
@maurolepore I think you also have to include the new argument `avoid_list_cols = TRUE` if we pivot wider. Isn't it?

Reply:
FYI: List columns are issue when we convert R dataframes into csv files.